components/EventAlbum/EventAlbum.tsx

```diff
@@ -1,14 +1,60 @@
 import Link from 'next/link';
 import { parseDate } from '@/app/util/DateTime';
-
+import { MeetupEvent } from '@/app/types/domain/MeetupEvent';
+import { getEventPhotoAlbum } from '@/app/service/meetup/MeetupService';
 /*
 This photo album component is designed specifically for the Previous events section 
 Parameters- 
 images : An array of objects containing image links and overlay text 
  */
 
-export default function EventAlbum({ pastEvents }: any) {
-  
+
+
+export default function EventAlbum({ pastEvents }: any) {  
+
+  const  getCoverImage = async (meetupEvent: any) => {
```

Comment:
Hmm this would be making making additional requests for each album just to grab the album cover... instead, this album cover data could be retrieved with the initial request

Reply:
Since I need to fetch the photos in the photo album to populate the covers, I will need to fetch at least the first 2 images per event.

Reply:
You can update the initial query to fetch all the data once, instead of making all of these secondary requests: 

https://github.com/CodeandCoffeeCommunity/boston-code-and-coffee/blob/main/app/service/meetup/graphql/fragments/GroupPastEventsFragment.ts#L16C19-L16C19

it would update this query to get the data you need

Reply:
> You can update the initial query to fetch all the data once, instead of making all of these secondary requests:
> 
> https://github.com/CodeandCoffeeCommunity/boston-code-and-coffee/blob/main/app/service/meetup/graphql/fragments/GroupPastEventsFragment.ts#L16C19-L16C19
> 
> it would update this query to get the data you need

Closing this pr. Will create another issue for this @alexpereira @Antosha9108 @IgorTheFixer